frontend/src/pages/DepartureDetails.tsx

```diff
@@ -40,97 +42,213 @@ const DeparturesDetails = () => {
     );
   }
 
-  const stopPosition = departure.destination.stop_position;
-  const startPosition = departure.departure.stop_position;
+  const routes = {
+    type: 'FeatureCollection',
+    features: departures.data
+      .filter((dep) => dep.geometry.length > 0)
+      .map((dep) => {
+        return {
+          type: 'Feature',
+          properties: {
+            color:
+              // todo: check <id> instead of <arrival_time> when <id> endpoint is available from api.
+              dep.departure.arrival_time === departure.departure.arrival_time
+                ? [19, 197, 123]
+                : [200, 200, 200],
+          },
+          geometry: { type: 'LineString', coordinates: dep.geometry },
+        };
+      })
+      .sort((a, b) => {
+        return a.properties.color[0] === 19
+          ? 1
+          : b.properties.color[0] === 19
+          ? -1
+          : 0;
+      }),
+  };
 
-  const geoJsonObject = {
+  const userRoute = {
     type: 'Feature',
     geometry: {
       type: 'LineString',
-      coordinates: departure?.geometry,
+      coordinates: [
+        [
+          parseFloat(fromTo.coordinates.lng),
+          parseFloat(fromTo.coordinates.lat),
+        ],
+        [
+          parseFloat(departure.departure.stop_position.lng),
+          parseFloat(departure.departure.stop_position.lat),
+        ],
+      ],
     },
   };
 
-  const busLayer = new GeoJsonLayer({
-    id: 'geojson-layer',
-    data: geoJsonObject,
+  const positions = departures.data
+    .filter((dep) => dep.departure.arrival_time && dep.destination.arrival_time)
+    .map((dep: Departure) => {
+      return {
+        userCoordinates: {
+          from: [
+            parseFloat(fromTo.coordinates.lng),
+            parseFloat(fromTo.coordinates.lat),
+          ],
+          to: [
+            parseFloat(dep.stops[0].stop_position.lng),
```

Comment:
This property more (easily accessible) exists on `dep.destination...`

Reply:
Fixed!